apps/sms/test/unit/contacts_test.js

```diff
@@ -467,56 +393,49 @@ suite('Contacts', function(done) {
 
         // navigator.mozContacts.find was called?
         assert.equal(mHistory.length, 1);
-        done();
-      });
+      }).then(done, done);
     });
 
     test('name first, part of email address last', function(done) {
       MockSettings.supportEmailRecipient = true;
       var mozContacts = navigator.mozContacts;
 
-        Contacts.findByString('Pepito a@b', function(contacts) {
-          done(function() {
-          var mHistory = mozContacts.mHistory;
+      Contacts.findByString('Pepito a@b').then(function(contacts) {
+        var mHistory = mozContacts.mHistory;
 
-          // contacts were found
-          assert.ok(Array.isArray(contacts));
-          assert.equal(contacts.length, 1);
+        // contacts were found
+        assert.ok(Array.isArray(contacts));
+        assert.equal(contacts.length, 1);
 
-          // navigator.mozContacts.find was called?
-          assert.equal(mHistory.length, 1);
-        });
-      });
+        // navigator.mozContacts.find was called?
+        assert.equal(mHistory.length, 1);
+      }).then(done, done);
     });
 
     test('part of email address first, name last', function(done) {
       MockSettings.supportEmailRecipient = true;
       var mozContacts = navigator.mozContacts;
 
-      Contacts.findByString('a@b Pepito', function(contacts) {
-        done(function() {
-          var mHistory = mozContacts.mHistory;
+      Contacts.findByString('a@b Pepito').then(function(contacts) {
+        var mHistory = mozContacts.mHistory;
 
-          // contacts were found
-          assert.ok(Array.isArray(contacts));
-          assert.equal(contacts.length, 1);
+        // contacts were found
+        assert.ok(Array.isArray(contacts));
+        assert.equal(contacts.length, 1);
 
-          // navigator.mozContacts.find was called?
-          assert.equal(mHistory.length, 1);
-        });
-      });
+        // navigator.mozContacts.find was called?
+        assert.equal(mHistory.length, 1);
+      }).then(done, done);
     });
 
     test('string search yields a contact without familyName', function(done) {
-      Contacts.findContactByString('julien 123', function(contacts) {
-        done();
-      });
+      Contacts.findContactByString('julien 123').then(() => done(), done);
```

Comment:
well, in the previous code, a similar case would not invoke the callback; with a rejection I think the same happens. So it would be quite similar here.

"catch" without effect is just like a `catch(e){}`: useless :)

I'm changing this